pkg/limits/ingest_limits.go

```diff
@@ -183,72 +183,110 @@ func (s *IngestLimits) stopping(failureCase error) error {
 	return failureCase
 }
 
-// GetLimits implements the IngestLimits service. It returns the current ingestion limits
-// for a given tenant, including stream count and stream rates.
-func (s *IngestLimits) GetLimits(_ context.Context, req *logproto.GetLimitsRequest) (*logproto.GetLimitsResponse, error) {
+// ServeHTTP implements the http.Handler interface.
+// It returns the current stream counts and status per tenant as a JSON response.
+func (s *IngestLimits) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
 	s.mtx.RLock()
 	defer s.mtx.RUnlock()
 
-	// Get streams for the requested tenant
-	streams, ok := s.metadata[req.Tenant]
-	if !ok {
-		return &logproto.GetLimitsResponse{
-			Limits: &logproto.IngestionLimits{
-				StreamCount: 0,
-				StreamRates: nil,
-			},
-		}, nil
+	// Get the cutoff time for active streams
+	cutoff := time.Now().Add(-s.cfg.WindowSize).UnixNano()
+
+	// Calculate stream counts and status per tenant
+	type recordedStream struct {
+		StreamHash uint64 `json:"streamHash"`
+		Active     bool   `json:"active"`
```

Comment:
I have a feeling this information is redundant. When would a distributor need to know about inactive streams?